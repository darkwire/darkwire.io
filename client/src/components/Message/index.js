import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Username from 'components/Username'
import moment from 'moment'
import Linkify from 'react-linkify'

class Message extends Component {
  render() {
    const msg = decodeURI(this.props.message)

    return (
      <div>
        <div className="chat-meta">
          <Username username={this.props.sender} />
          <span className="muted timestamp">
            {moment(this.props.timestamp).format('LT')}
          </span>
        </div>
        <div className="chat">
          <Linkify properties={{
            target: '_blank',
            rel: 'noopener noreferrer',
          }}>{msg}</Linkify>
        </div>
      </div>
    )
  }
}

Message.propTypes = {
  sender: PropTypes.string.isRequired,
  timestamp: PropTypes.number.isRequired,
  message: PropTypes.string.isRequired,
}

export default Message
