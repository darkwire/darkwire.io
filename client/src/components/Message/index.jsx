import PropTypes from 'prop-types';
import moment from 'moment';
import Linkify from 'react-linkify';

import Username from '@/components/Username';

const Message = ({ message, timestamp, sender }) => {
  const msg = decodeURI(message);

  return (
    <div>
      <div className="chat-meta">
        <Username username={sender} />
        <span className="muted timestamp">{moment(timestamp).format('LT')}</span>
      </div>
      <div className="chat">
        <Linkify
          properties={{
            target: '_blank',
            rel: 'noopener noreferrer',
          }}
        >
          {msg}
        </Linkify>
      </div>
    </div>
  );
};

Message.propTypes = {
  sender: PropTypes.string.isRequired,
  timestamp: PropTypes.number.isRequired,
  message: PropTypes.string.isRequired,
};

export default Message;
