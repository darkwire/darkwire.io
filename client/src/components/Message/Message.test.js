import React from 'react';
import { render } from '@testing-library/react';
import Message from '.';

test('Message component is displaying', async () => {
  const { asFragment } = render(<Message sender={'linus'} timestamp={1588794269074} message={'we come in peace'} />);

  expect(asFragment()).toMatchSnapshot();
});
