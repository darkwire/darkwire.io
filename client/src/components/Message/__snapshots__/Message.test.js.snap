// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message component is displaying 1`] = `
<DocumentFragment>
  <div>
    <div
      class="chat-meta"
    >
      <span
        class="username"
        style="color: rgb(131, 239, 135);"
      >
        linus
      </span>
      <span
        class="muted timestamp"
      >
        7:44 PM
      </span>
    </div>
    <div
      class="chat"
    >
      <span
        class="Linkify"
      >
        we come in peace
      </span>
    </div>
  </div>
</DocumentFragment>
`;
